dhall/src/Dhall/Parser/Expression.hs

```diff
@@ -293,9 +293,21 @@ parsers embedded = Parsers {..}
         ]
 
     applicationExpression = do
-            f <-    (Some <$ try (_Some <* nonemptyWhitespace))
-                <|> return id
-            a <- noted importExpression_
+            -- Here f can be `Some`
+            let alternative0 = do
+                    f <- (Some <$ try (_Some <* nonemptyWhitespace))
+
+                    return (f, True)
+
+            -- Or something else like `x ===`
```

Comment:
```suggestion
            -- Or something else like `Natural/even`
```

Technically, `x ===` would be elsewhere within the parser for parsing operators.  `applicationExpression` is only used to parse ordinary function applications

Reply:
Ok, it's just that `Natural/even` doesn't cause a parser failure... I'll keep the `x ===` in mind when adding error message test.